sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -785,7 +785,23 @@ public User getUser() {
   private String getAndroidId() {
     // Android 29 has changed and -> Avoid using hardware identifiers, find another way in the
     // future
-    return Settings.Secure.getString(context.getContentResolver(), Settings.Secure.ANDROID_ID);
+    String androidId =
+        Settings.Secure.getString(context.getContentResolver(), Settings.Secure.ANDROID_ID);
+
+    //    https://android-developers.googleblog.com/2011/03/identifying-app-installations.html
+    if (androidId == null
+        || androidId.isEmpty()
+        || androidId.toLowerCase(Locale.ROOT).contentEquals("9774d56d682e549c")) {
+      try {
+        androidId = Installation.id(context);
```

Comment:
This code is now potentially doing blocking I/O on the calling thread (which might be the UI thread).

Reply:
yeah, it is, I'm gonna fix it together with all the other IO calls later on.

Reply:
btw this will only be executed for a very small portion of the devices, the bug happens only for a specific brand and API version.